forums/views.py

```diff
@@ -33,3 +35,70 @@ def user_login(request):
 def user_logout(request):
     logout(request)
     return HttpResponseRedirect('/')
+
+def forgotpassword(request):
+    context = {}
+    user_emails = []
+    context.update(csrf(request))
+    if request.method == 'POST':
+        users = User.objects.all()
+        for user in users:
+            user_emails.append(user.email)
+        email = request.POST['email']
+        if email == "":
+            context['invalid_email'] = True
+            return render_to_response("forums/templates/forgot-password.html", context)
+        if email in user_emails:
+            user = User.objects.get(email=email)
+            password = ''.join(random.choice(string.ascii_uppercase + string.digits) for _ in range(8))
+            user.set_password(password)
+            user.save()
+            sender_name = "Spoken Forums"
+            sender_email = "no-reply@spoken-tutorial.org"
+            subject = "Spoken Forums - Password Reset"
+            to = (user.email, )
+	    url = settings.EMAIL_URL
+            message = """Dear """+user.username+""",\nYour password for Spoken Forums has been reset. Your credentials are:\nUsername: """+user.username+"""\nPassword: """+password+"""\n\nWe recommend you to login with the given credentials & update your password immediately.\nLink to set new password: """+url+"""/accounts/login/?next=/accounts/update-password/\nThank You !\nRegards,\nSpoken Team,\n IIT Bombay."""
```

Comment:
Please use proper forgot password flow and let the user choose a password instead of generating and sending one. The entropy of the password here is 8 digit with ascii_uppercase and quite easy to guess for somebody. 

This can't go in, as it has a security issue. If you prefer this feature can come as a different pull-request. Also use the in-built views django provides for forgot password functionality instead of make own.

Reply:
This is already on server. was showing in modified files. I just committed it to start my work